maven-core/src/main/java/org/apache/maven/internal/impl/DefaultArtifactCoordinate.java

```diff
@@ -30,11 +31,11 @@
  * A wrapper class around a maven resolver artifact.
  */
 public class DefaultArtifactCoordinate implements ArtifactCoordinate {
-    private final @Nonnull AbstractSession session;
```

Comment:
Can this change be removed from this PR ? It's completely unrelated. 
Also, if we do change that, other classes such as `DefaultArtifact` and maybe others use the same mechanism and should be changed at the same time.

Reply:
Sure! I've just pushed a commit to undo this part.